src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -455,9 +456,18 @@ public function formAction($id, Request $request)
         }
 
         $productAdapter = $this->get('prestashop.adapter.data_provider.product');
-        $product = $productAdapter->getProduct($id);
+        try {
+            $product = $productAdapter->getProduct($id);
+        } catch (LogicException $e) {
```

Comment:
Looks weird to catch a LogicException instead of a ProductNotFoundException

Reply:
getProduct method throws that exception (don't ask - legacy stuff.. :smile:), to avoid introducing BC breaks or other unexpected behavior (modules may depend on that exception) we suggested not to change the legacy part and just catch that one.